### DISCUSSION TITLE
Option to auto create Umbraco Database

### DISCUSSION THREAD

**Discussion by royberris** [umbraco/Umbraco-CMS#15294] [Features and ideas]
When working with Umbraco and SQL Server we need to manually create a database before we can connect to it. When working with containerized databases in development this is quite annoying because it requires a bulky auto create script when starting up the container. Same thing for cloud databases.

I would like to see an option where Umbraco can create a database if it does not exists when the given user has the rights.

When working with EF Core for example we have the option to run migrations on startup, which will automatically create a database if it does not exist. Or we can use the `EnsureCreated` on a database. Here it is also used to create all tables.
https://learn.microsoft.com/en-us/ef/core/managing-schemas/ensure-created#ensurecreated

For Umbraco it would be a nice step when running the installer for example? Allowing us to check a box to create the database. Or an option when using unattended install.

**Comment by rasmusjp** [comment]
Hi @royberris

This should already be possible if you set `Umbraco:CMS:Global:InstallMissingDatabase=true` and the connection string has `CREATE DATABASE` permission, see https://docs.umbraco.com/umbraco-cms/fundamentals/setup/install/unattended-install and https://docs.umbraco.com/umbraco-cms/reference/configuration/globalsettings#install-missing-database

**Reply by royberris** [reply]
Wow, didn't know this.. I tried looking for it but must be blind. Thanks!
